projects/angular-l10n/src/lib/directives/l10n-date.directive.ts

```diff
@@ -25,7 +26,7 @@ export class L10nDateDirective extends L10nDirective {
     }
 
     protected getValue(text: string): string {
-        return this.intl.formatDate(text, this.options);
+        return this.intl.formatDate(text, this.options, this.useDateLanguage);
```

Comment:
So, the reason behind this is following.
If you use English for translations and German for dates and you have a date directive that is displaying a date with `[options]="{dateStyle: 'full'}"` you end up with date labels (day and month) displayed in German and the rest of the web app is in English. Hence, I wanted a way to decide which locale to use for the dates.
First I decided to just use `@Input() language` in the directive, but then we might have the following problem.
The base `L10nDirective` is listening for the translation changes and the user is passing an input with the language, `<p [language]="locale.dateLanguage" [options]="{ dateStyle: 'full' }" l10nDate>{{ today }}</p>`. If the `L10nDirective` is faster, then the wrong language will be used to format the date.
What would be your approach here?

Reply:
Adding the language property to directives (and also to validation directives and async pipes) would be great, because all components would be aligned and would allow you to manage multilingual components!

So something like this:
```
export class L10nDateDirective extends L10nDirective {
    [...]

    @Input() public language: string;

    [...]
    protected getValue(text: string): string {
        return this.intl.formatDate(text, this.options, this.language || undefined);
    }
}
```
should work, because:

case 1) If the `ngOnChanges` method of the `L10nDirective` https://github.com/robisim74/angular-l10n/blob/a42ef6499300cb85b6674d3a6ddab63db4f4e7bc/projects/angular-l10n/src/lib/models/l10n-directive.ts#L39 is fired before the `addTranslationListener` method, https://github.com/robisim74/angular-l10n/blob/a42ef6499300cb85b6674d3a6ddab63db4f4e7bc/projects/angular-l10n/src/lib/models/l10n-directive.ts#L112 `getValue` will update using the new language

case 2) If the `ngOnChanges` method is fired after, `getValue `will use the new language

Ok, it is fired twice! We should find a way not to turn it on twice.

Reply:
Will this mean that the language property needs to be mandatory?
In this case, we can get rid of the `addTranslationListener` and just rely on `ngOnChanges` I guess. This will also help with the problem `getValue` being called twice.

But this seems like a really nasty breaking change. The developer will need to pass the language to all directives.

Reply:
No, the language property would not be mandatory. If one decides to pass the new language, that would be used, otherwise everything would continue to work as before.

Reply:
What do you think about this solution there (prevents calling the `getValue` twice)?

Inside the base `L10nDirective`.

`
@Input() language: string;

public ngAfterViewInit(): void {
        if (this.el && this.el.nativeElement) {
            this.element = this.el.nativeElement;
            this.renderNode = getTargetNode(this.element);
            this.text = this.getText();
            this.attributes = this.getAttributes();
            this.addTextListener();

            **if ( this.language ) {
                this.replaceText()
                this.replaceAttributes();
            } else {
                this.addTranslationListener();
            }**
        }
    }
`

We don't need to register the translation listener if the language prop exists. `ngOnChanges` will take care of `replaceTexet()` and `replaceAttributes`?

Reply:
Yes, it should work properly, because `ngOnChanges `activates when properties change value (and it calls the replace methods)